libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F4/serial_api.c

```diff
@@ -141,18 +235,18 @@ void serial_init(serial_t *obj, PinName tx, PinName rx)
     }
 
     // Configure UART
-    obj->baudrate = 9600;
-    obj->databits = UART_WORDLENGTH_8B;
-    obj->stopbits = UART_STOPBITS_1;
-    obj->parity   = UART_PARITY_NONE;
+    _SERIAL_OBJ(baudrate) = 9600;
```

Comment:
Hi, I'll rename the macro and check your other comments below. 
You're right, this is needed as long as this feature is not deployed on every F4.
Cheers